app/src/ui/lib/action-status-icon.tsx

```diff
@@ -1,40 +1,42 @@
 import * as React from 'react'
-import { Octicon, OcticonSymbol } from '../octicons'
-import { assertNever } from '../../lib/fatal-error'
 import * as classNames from 'classnames'
-import { MergeResult } from '../../models/merge'
 import { ComputedAction } from '../../models/computed-action'
+import { assertNever } from '../../lib/fatal-error'
+
+import { Octicon, OcticonSymbol } from '../octicons'
+
+interface IActionStatusIconProps {
+  /** The status to display to the user */
+  readonly status: { kind: ComputedAction } | null
+
+  /** A required class name prefix for the Octicon component */
+  readonly classNamePrefix: string
 
-interface IMergeStatusIconProps {
   /** The classname for the underlying element. */
   readonly className?: string
-
-  /** The status to display. */
-  readonly status: MergeResult | null
 }
 
-/** The little CI status indicator. */
-export class MergeStatusHeader extends React.Component<
-  IMergeStatusIconProps,
-  {}
-> {
+/** An indicator to display representing the result of a computed action */
```

Comment:
This documentation isn't providing me with enough information to understand what this component is doing and given that the name is rather opaque as well I think we need to invest a little more time in a thorough description here. I think this class could do with a rename to `ComputedActionIcon` or something similar to make the intentions even more clear that it's not a general-purpose component.

Having read through the previous PRs to figure out what a `ComputedAction` is I think that too could do with a bit of clarification but that's obviously outside of the scope of this PR.

```suggestion
/** 
 * A component used to render a visual indication of a `ComputedAction` state.
 *
 * In essence this is a small wrapper around an `Octicon` which determines which
 * icon to use based on the `ComputedAction`. A computed action is essentially
 * the current state of a merge or a rebase operation and this component is used
 * in the header of merge or rebase conflict dialogs to augment the textual
 * representation of the current merge or rebase progress.
 */
```